tests/test_earth_radius.F90

```diff
@@ -0,0 +1,91 @@
+program test_earth_radius
+  use earth_radius_mod
+  implicit none
+
+  integer :: igdtlen
+  parameter(igdtlen = 7)
+  integer :: igdtmpl(igdtlen)
+  real :: eccen_squared
+  real :: radius
+  
+  print *,'Testing earth_radius_mod...'
+
+  ! Case 0.
+  igdtmpl(1) = 0
+  call earth_radius(igdtmpl, igdtlen, radius, eccen_squared)
+  if (abs(radius - 6367470.0) > epsilon(radius)) stop 2
+  if (abs(eccen_squared - 0.0) > epsilon(eccen_squared)) stop 3
+
+  ! Case 1.
+  igdtmpl(1) = 1
+  igdtmpl(2) = 1
+  igdtmpl(3) = 1
```

Comment:
igdtmpl(3) contains octets 17-20, which holds the scale value of the radius of a spherical earth.  I think your example produces an earth radius of 1 meter. I know these are idealized tests. But I might pick a more realistic earth radius.

Reply:
Good idea!